### ISSUE TITLE
[Bug] Public link site: remove "New" btn

### ISSUE THREAD

**Issue by tbsbdr** [owncloud/web#5618]
## Problem

- **New button** If I receive a public link with read-only permissions, the new-Button is still shown on the public link site.
~~- **Context menu** no context menu is shown on righclick or 3-dots menu~~

### Steps to reproduce
1. login to https://ocis.ocis-web.latest.owncloud.works
2. upload a file
3. create a public link for this file with role **"viewer"**
4. open this link in a private window
5. "new" button is displayed, ~~no context menu showing~~
![2021-08-03_17-59-00 (1)](https://user-images.githubusercontent.com/26610733/128047932-8be7837b-3f5b-4df1-bc35-2027adb4b5ba.gif)

### Expected behaviour
- hide the New button, if you dont have write-permission
- no jumpy layout when selecting / deselecting files (batch actions become visible)
- ~~provide context menu via 3-dots menu~~
- ~~provide context menu via rightclick menu~~

### Actual behaviour
~~- click on 3-dots menu: noting happens~~
- righclick on file: shows browser-context menu

**Comment by pascalwengerter** [comment]
Can't reproduce the nonexistant context-menu on the table row. The 3-dot menu bug should be fixed with the next ODS release

**Comment by tbsbdr** [comment]
"The 3-dot menu bug" does not exist any more, thx for notice! updated description + title

**Comment by dschmidt** [comment]
The New button is disabled on readonly public links because `canUpload` is `false`. 

https://github.com/owncloud/web/blob/7332f220d1b415e06656ff281d68829122151cfa/packages/web-app-files/src/components/AppBar/AppBar.vue#L276

...but you want the button to be completely removed? Only on public links or whenever nothing can be created?
Only when `canUpload` is `false` or also when there is no free space?

**Comment by kulmann** [comment]
Yes, public links should never shows the button at all if permissions for creating files/folders are not given, while authenticated contexts should show it disabled for the same situation. The idea is, that for public links the permission will not change for subfolders and therefore it's confusing to see the button at all. In an authenticated context it could change depending on the folder you're in - which would result in a "jumpy" ui when navigating through your folder structure.

TL;DR:
- public link without permissions for creating files/folders: hide button
- authenticated context in a folder without permissions for creating files/folders: disable button

**Comment by dschmidt** [comment]
Okay, great - thanks for the clarification!
